### PR TITLE
fix: set request top_k, needed to be read by jinabox

### LINKED ISSUE #942: Not able to interpet top_k introduced by jinabox (And I guess also not by python client)
**Describe the bug**
Currently `jina` is not able to pass and interpret the `top_k` parameter sent by `jinabox`. 

PR #941 fixes the issue 

┆Issue is synchronized with this [Jira Task](https://jinaai.atlassian.net/browse/JINACORE-229) by [Unito](https://www.unito.io/learn-more)

### PR DESCRIPTION
**Pull request by JoanFM**
Not working because SearchRequest does not have `top_k`, so how do we interpeet the message from `jinabox`?

Fixes #942

### CODE REVIEW

File: jina/clients/python/request.py [hunk 1]
```diff
@@ -87,6 +87,16 @@ def _generate(data: Union[Iterator['jina_pb2.Document'], Iterator[bytes], Iterat
                 queryset = [queryset]
             req.queryset.extend(queryset)
 
+        if top_k and mode == ClientMode.SEARCH:
+            if top_k <= 0:
+                raise ValueError(f'"top_k: {top_k}" is not a valid number')
```
Comment (maximilianwerk) [inline jina/clients/python/request.py:92]: I am not sure about the context of this exception, but I read it as: The client can define a value, which will make jina throw an exception. Is this exception catches on a per-request basis? If so, everything is fine. If not, a client can crash jina with a bad request. That would be not good.

**Comment by maximilianwerk** [comment]
Just tested on the freshly refactored southpark example and it worked. :+1:

**Comment by maximilianwerk** [comment]
LGTM

**Review by maximilianwerk — APPROVED** [review]
LGTM

File: jina/clients/python/request.py (hunk 1 above)
Comment (JoanFM) [reply jina/clients/python/request.py:92]: This works on the client, so jina would, so not necessarily. This exception is taken from an old version where this functionality was present.

**Pull request merged**